backend/portal/api/curation/v1/curation/collections/collection_id/datasets/dataset_id/actions.py

```diff
@@ -1,41 +1,161 @@
+                dataset["processing_status_detail"] = d.status.validation_message
+                dataset["processing_status"] = "VALIDATION_FAILURE"
+            else:
+                dataset["processing_status"] = "PIPELINE_FAILURE"
+        else:
+            dataset["processing_status"] = d.status.processing_status
+
+    dataset_ontology_elements = DATASET_ONTOLOGY_ELEMENTS_PREVIEW if preview else DATASET_ONTOLOGY_ELEMENTS
+    for ontology_element in dataset_ontology_elements:
+        if dataset_ontology_element := dataset.get(ontology_element):
+            if not isinstance(dataset_ontology_element, list):
+                # Package in array
+                dataset[ontology_element] = [dataset_ontology_element]
+        else:
+            dataset[ontology_element] = []
+
+    if not preview:  # Add these fields only to full (and not preview) Dataset metadata response
+        dataset["revision_of"] = d.canonical_dataset.dataset_id.id
+        dataset["title"] = d.metadata.name
+        if d.metadata.is_primary_data is not None:
+            dataset["is_primary_data"] = is_primary_data_mapping.get(d.metadata.is_primary_data, [])
+
+    return dataset
+
+
 def get(collection_id: str, dataset_id: str = None):
-    db_session = g.db_session
-    if not db_session.query(DbCollection.id).filter(DbCollection.id == collection_id).first():
+    business_logic = get_business_logic()
+
+    collection_version = business_logic.get_collection_version_from_canonical(CollectionId(collection_id))
+    if collection_version is None:
         raise NotFoundHTTPException("Collection not found!")
-    dataset = get_dataset_else_error(db_session, dataset_id, collection_id)
-    response_body = reshape_dataset_for_curation_api(dataset.to_dict_keep(EntityColumns.columns_for_dataset))
+
+    version = get_infered_dataset_version(dataset_id)
+    if version is None:
+        raise NotFoundHTTPException("Dataset not found")
+
+    response_body = _reshape_dataset_for_curation_api(version)
     return make_response(jsonify(response_body), 200)
 
 
-@dbconnect
 def delete(token_info: dict, collection_id: str, dataset_id: str = None):
-    db_session = g.db_session
-    dataset = get_dataset_else_error(db_session, dataset_id, collection_id, include_tombstones=True)
-    delete_dataset_common(db_session, dataset, token_info)
-    return "", 202
+
+    business_logic = get_business_logic()
+    user_info = UserInfo(token_info)
+
+    # TODO: deduplicate from ApiCommon. We need to settle the class/module level debate before can do that
+    dataset_version = business_logic.get_dataset_version(DatasetVersionId(dataset_id))
+    if dataset_version is None:
+        raise ForbiddenHTTPException(f"Dataset {dataset_id} does not exist")
+
+    collection_version = business_logic.get_collection_version(CollectionVersionId(collection_id))
+    # If the collection does not exist, it means that the dataset is orphaned and therefore we cannot
```

Comment:
I think this isn't supposed to happen, but I added it anyway. For what it's worth, `get_collection_version` returns an `Optional` so pylance will force you to add the `if is not None` condition, or you'll get a warning.

Reply:
Is this a condition we can test for?

Reply:
I don't think it's necessary. Also the test should rely on a database state that's impossible to achieve. If you prefer, we can add a TODO to double check or specify that the check is only to prevent a possible edge case that should not be possible in the system.

Reply:
Actually nvm, this was a comment taken from the portal API. In this case, this will simply be raised if a wrong collection_id is specified (i.e., a collection_id that doesn't contain the specified dataset_id). I removed the comment.